pxtarget.json

```diff
@@ -15,7 +15,22 @@
         "publishing": true,
         "githubPackages": true,
         "cloudProviders": {
-            "github": {}
+            "github": {
+                "id": "github",
+                "name": "GitHub"
+            },
+            "microsoft": {
+                "id": "microsoft",
+                "name": "Microsoft",
+                "identity": true,
+                "redirect": true
```

Comment:
What does redirect mean here? And why doesn't github need it?

Reply:
`redirect` was an existing field in `AppCloudProvider`, here: https://github.com/microsoft/pxt/blob/541d9c867beaed9029aefb2cf65a3c4bcb1efaba/localtypings/pxtarget.d.ts#L174. If set to true, use "redirect" auth flow as opposed to "popup". The popup flow is discouraged these days, so I'm explicitly configuring it to use redirect. Not referencing it in code yet though.